core/lib/network-request.js

```diff
@@ -124,11 +124,25 @@ class NetworkRequest {
     this.parsedURL = /** @type {ParsedURL} */ ({scheme: ''});
     this.documentURL = '';
 
-    this.startTime = -1;
-    /** @type {number} */
-    this.endTime = -1;
-    /** @type {number} */
-    this.responseReceivedTime = -1;
+    /** ms. When the main thread initially discovers a network request. */
+    this.mainThreadStartTime = -1;
+    /**
+     * ms. When the network service is about to handle a request, ie. just before going to the
+     * HTTP cache or going to the network for DNS/connection setup.
+     */
+    this.networkRequestTime = -1;
+    /** ms. When the first byte of the response headers is received. */
+    this.responseHeadersReceivedTime = -1;
```

Comment:
bikeshed: I think I prefer the old `responseReceivedTime` name (with clarification of how it's defined in a comment).

I can see the benefits of being clearer internally, but when using an instance externally, this property name makes me have to think a lot to parse it, and it kind of implies there are other response-received times, even though there's just the one :)

Reply:
(i'd prefer `headersReceivedTime` to be more terse)

do you think we'd ever want to introduce more granular network timings? if so, we should be specific here

Reply:
maybe we can drop "headers" if we sneak "start" into the name somewhere.

* `responseStartTime` i like. i guess we could just call out the 'headers received' technicality in a comment as the distinction isn't really important.
* `responseHeadersReceivedTime` does feel a bit wordy and overkill at this point.
* `headersReceivedTime` well... less wordy but it gets me thinking about "but what about request headers??"  so the thinking overhead is still there.
* `responseReceivedTime` (the old one) i dont like cuz it can easily be interpreted as 'end of response received'

Reply:
`responseHeadersEndTime` is another option if we want to be both specific, consistent with start/end, and precise, though it's unfortunate that usually what we want is `responseBodyStartTime`.

`responseStartTime` does seem like it would trip up a lot on the fact that the server has already responded :/